test/scholar/linear/svm_test.exs

```diff
@@ -0,0 +1,186 @@
+        [6.6, 3.0, 4.4, 1.4],
+        [6.8, 2.8, 4.8, 1.4],
+        [6.7, 3.0, 5.0, 1.7],
+        [6.0, 2.9, 4.5, 1.5],
+        [5.7, 2.6, 3.5, 1.0],
+        [5.5, 2.4, 3.8, 1.1],
+        [5.5, 2.4, 3.7, 1.0],
+        [5.8, 2.7, 3.9, 1.2],
+        [6.0, 2.7, 5.1, 1.6],
+        [5.4, 3.0, 4.5, 1.5],
+        [6.0, 3.4, 4.5, 1.6],
+        [6.7, 3.1, 4.7, 1.5],
+        [6.3, 2.3, 4.4, 1.3],
+        [5.6, 3.0, 4.1, 1.3],
+        [5.5, 2.5, 4.0, 1.3],
+        [5.5, 2.6, 4.4, 1.2],
+        [6.1, 3.0, 4.6, 1.4],
+        [5.8, 2.6, 4.0, 1.2],
+        [5.0, 2.3, 3.3, 1.0],
+        [5.6, 2.7, 4.2, 1.3],
+        [5.7, 3.0, 4.2, 1.2],
+        [5.7, 2.9, 4.2, 1.3],
+        [6.2, 2.9, 4.3, 1.3],
+        [5.1, 2.5, 3.0, 1.1],
+        [5.7, 2.8, 4.1, 1.3],
+        [6.3, 3.3, 6.0, 2.5],
+        [5.8, 2.7, 5.1, 1.9],
+        [7.1, 3.0, 5.9, 2.1],
+        [6.3, 2.9, 5.6, 1.8],
+        [6.5, 3.0, 5.8, 2.2],
+        [7.6, 3.0, 6.6, 2.1],
+        [4.9, 2.5, 4.5, 1.7],
+        [7.3, 2.9, 6.3, 1.8],
+        [6.7, 2.5, 5.8, 1.8],
+        [7.2, 3.6, 6.1, 2.5],
+        [6.5, 3.2, 5.1, 2.0],
+        [6.4, 2.7, 5.3, 1.9],
+        [6.8, 3.0, 5.5, 2.1],
+        [5.7, 2.5, 5.0, 2.0],
+        [5.8, 2.8, 5.1, 2.4],
+        [6.4, 3.2, 5.3, 2.3],
+        [6.5, 3.0, 5.5, 1.8],
+        [7.7, 3.8, 6.7, 2.2],
+        [7.7, 2.6, 6.9, 2.3],
+        [6.0, 2.2, 5.0, 1.5],
+        [6.9, 3.2, 5.7, 2.3],
+        [5.6, 2.8, 4.9, 2.0],
+        [7.7, 2.8, 6.7, 2.0],
+        [6.3, 2.7, 4.9, 1.8],
+        [6.7, 3.3, 5.7, 2.1],
+        [7.2, 3.2, 6.0, 1.8],
+        [6.2, 2.8, 4.8, 1.8],
+        [6.1, 3.0, 4.9, 1.8],
+        [6.4, 2.8, 5.6, 2.1],
+        [7.2, 3.0, 5.8, 1.6],
+        [7.4, 2.8, 6.1, 1.9],
+        [7.9, 3.8, 6.4, 2.0],
+        [6.4, 2.8, 5.6, 2.2],
+        [6.3, 2.8, 5.1, 1.5],
+        [6.1, 2.6, 5.6, 1.4],
+        [7.7, 3.0, 6.1, 2.3],
+        [6.3, 3.4, 5.6, 2.4],
+        [6.4, 3.1, 5.5, 1.8],
+        [6.0, 3.0, 4.8, 1.8],
+        [6.9, 3.1, 5.4, 2.1],
+        [6.7, 3.1, 5.6, 2.4],
+        [6.9, 3.1, 5.1, 2.3],
+        [5.8, 2.7, 5.1, 1.9],
+        [6.8, 3.2, 5.9, 2.3],
+        [6.7, 3.3, 5.7, 2.5],
+        [6.7, 3.0, 5.2, 2.3],
+        [6.3, 2.5, 5.0, 1.9],
+        [6.5, 3.0, 5.2, 2.0],
+        [6.2, 3.4, 5.4, 2.3],
+        [5.9, 3.0, 5.1, 1.8]
+      ])
+
+    x = Scholar.Preprocessing.standard_scale(x, axes: [-1])
+    y = Nx.concatenate([Nx.broadcast(0, {50}), Nx.broadcast(1, {50}), Nx.broadcast(2, {50})])
+
+    shuffle = Nx.iota({Nx.axis_size(x, 0)})
+    {shuffle, _} = Nx.Random.shuffle(key, shuffle)
+    x = Nx.take(x, shuffle)
+    y = Nx.take(y, shuffle)
+    {x_train, x_test} = Nx.split(x, 120)
+    {y_train, y_test} = Nx.split(y, 120)
+    {x_train, x_test, y_train, y_test}
+  end
+
+  test "Iris Data Set - multinomial classification svm test" do
+    {x_train, x_test, y_train, y_test} = iris_data()
+
+    model = SVM.fit(x_train, y_train, num_classes: 3, margin: 150)
```

Comment:
keeping in with the parameterized loss: `:margin` and `:c` should be loss-specific opts instead, generally enclosed in a function capture: `SVM.hinge_loss(&1, &2, c: c, margin: margin)`